src/NugetFeedDeploymentRepository/NugetPackageExtractor.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Packaging.PackageExtraction;
+
+namespace Etg.Yams.NuGet.Storage
+{
+    public class NugetPackageExtractor
```

Comment:
Looks like a utils, or even just a private method inside the nuget repository

Reply:
I wanted to keep it separate from the repository so it can be easily reused in the future. I was thinking about supporting a scenario where the packages are downloaded from blob storage instead of nuget feed.

Reply:
Ok no problem. I'd consider putting it behind an interface in that case (it's more than a simple POCO).